File: TimberCruisingApp/Screens/ProjectDashboardScreen.swift
// Spec §3.1 REQ-PRJ-002/003/004. Project dashboard — strata list, planned-plot
// summary, entry points into CruiseDesign / PlotMap / Export / Settings.
//
// Phase 7.4 redesign: whole screen now reads top-to-bottom as a
// step-by-step guide (①→②→③→④). Each step shows its own friendly
// description and a primary action. The strata step supports both
// "draw on map" (no file needed) and legacy "import from file" paths.

import SwiftUI
import Models
#if canImport(UniformTypeIdentifiers)
import UniformTypeIdentifiers
#endif

public struct ProjectDashboardScreen: View {

    @EnvironmentObject private var environment: AppEnvironment
    @StateObject private var viewModel: ProjectDashboardViewModel
    @State private var isPresentingImporter = false
    @State private var importFormat: ProjectDashboardViewModel.ImportFormat = .geoJSON

    public init(project: Project) {
        _viewModel = StateObject(wrappedValue: ProjectDashboardViewModel(project: project))
    }

    public var body: some View {
        List {
            gettingStartedSection
            summarySection
            strataSection
            planSection
            cruiseSection
            toolsSection
        }
        .navigationTitle(viewModel.project.name)
        #if os(iOS)
        .navigationBarTitleDisplayMode(.inline)
        #endif
        #if canImport(UniformTypeIdentifiers)
        .fileImporter(
            isPresented: $isPresentingImporter,
            allowedContentTypes: allowedTypes,
            allowsMultipleSelection: false
        ) { result in
            guard case .success(let urls) = result, let url = urls.first else { return }
            let accessed = url.startAccessingSecurityScopedResource()
            defer { if accessed { url.stopAccessingSecurityScopedResource() } }
            viewModel.importStrata(fileURL: url, format: importFormat)
        }
        #endif
        .task {
            viewModel.configure(with: environment)
            viewModel.refresh()
        }
        .alert("Something went wrong",
               isPresented: Binding(
                    get: { viewModel.errorMessage != nil },
                    set: { if !$0 { viewModel.errorMessage = nil } })
        ) {
            Button("OK", role: .cancel) { viewModel.errorMessage = nil }
        } message: {
            Text(viewModel.errorMessage ?? "")
        }
    }

    // MARK: - Getting started (progress guide)

    @ViewBuilder
    private var gettingStartedSection: some View {
        Section {
            VStack(alignment: .leading, spacing: 10) {
                Text("How this works")
                    .font(.headline)
                stepRow(n: 1, done: !viewModel.strata.isEmpty,
                        title: "Define strata",
                        hint: "Draw boundaries on the map, or import GeoJSON / KML.")
                stepRow(n: 2, done: viewModel.design != nil && !viewModel.plannedPlots.isEmpty,
                        title: "Design cruise + generate plots",
                        hint: "Pick plot size and spacing — sample plots are generated automatically.")
                stepRow(n: 3, done: false,
                        title: "Measure in the field (Go Cruise)",
                        hint: "Walk to each plot and measure trees one by one. DBH via LiDAR, height via AR.")
                stepRow(n: 4, done: false,
                        title: "Review + export",
                        hint: "Check stand statistics and export to PDF / CSV / GeoJSON.")
            }
            .padding(.vertical, 4)
        } header: {
            Text("Guide")
        }
    }

    private func stepRow(n: Int, done: Bool, title: String, hint: String) -> some View {
        HStack(alignment: .top, spacing: 10) {
            ZStack {
                Circle()
                    .fill(done ? Color.green : Color.accentColor.opacity(0.15))
                    .frame(width: 28, height: 28)
                if done {
                    Image(systemName: "checkmark")
                        .font(.caption.bold())
                        .foregroundStyle(.white)
                } else {
                    Text("\(n)")
                        .font(.caption.bold())
                        .foregroundStyle(Color.accentColor)
                }
            }
            VStack(alignment: .leading, spacing: 2) {
                Text(title).font(.subheadline.bold())
                Text(hint).font(.caption).foregroundStyle(.secondary)
                    .fixedSize(horizontal: false, vertical: true)
            }
            Spacer()
        }
    }

    // MARK: - Summary

    private var summarySection: some View {
        Section("Summary") {
            LabeledContent("Units", value: viewModel.project.units.rawValue.capitalized)
            LabeledContent("Total area", value: formatAcres(viewModel.totalAcres))
            LabeledContent("Strata", value: "\(viewModel.strata.count)")
            LabeledContent("Planned plots", value: "\(viewModel.plannedPlots.count)")
        }
    }

    // MARK: - Strata (step 1)

    @ViewBuilder
    private var strataSection: some View {
        Section {
            if viewModel.strata.isEmpty {
                VStack(alignment: .leading, spacing: 10) {
                    Label("No strata yet", systemImage: "map")
                        .font(.subheadline.bold())
                    Text("A stratum is a cutting block you want to measure. Draw corners on the map, or import a prepared GeoJSON / KML file.")
                        .font(.caption)
                        .foregroundStyle(.secondary)
                        .fixedSize(horizontal: false, vertical: true)

                    NavigationLink {
                        StratumDrawScreen(project: viewModel.project)
                    } label: {
                        Label("Draw stratum on map", systemImage: "pencil.and.outline")
                            .frame(maxWidth: .infinity, minHeight: 44)
                    }
                    .buttonStyle(.borderedProminent)
                    .accessibilityIdentifier("dashboard.drawStratum")

                    Menu {
                        Button("Import GeoJSON") {
                            importFormat = .geoJSON
                            isPresentingImporter = true
                        }
                        Button("Import KML") {
                            importFormat = .kml
                            isPresentingImporter = true
                        }
                    } label: {
                        Label("Import from file", systemImage: "square.and.arrow.down")
                            .frame(maxWidth: .infinity, minHeight: 36)
                    }
                    .buttonStyle(.bordered)
                    .accessibilityIdentifier("dashboard.importMenu")
                }
                .padding(.vertical, 4)
            } else {
                ForEach(viewModel.strata) { stratum in
                    HStack {
                        VStack(alignment: .leading, spacing: 2) {
                            Text(stratum.name)
                            Text(formatAcres(Double(stratum.areaAcres)))
                                .font(.caption)
                                .foregroundStyle(.secondary)
                        }
                        Spacer()
                    }
                }
                .onDelete { idx in
                    for i in idx { viewModel.delete(stratumId: viewModel.strata[i].id) }
                }
                NavigationLink {
                    StratumDrawScreen(project: viewModel.project)
                } label: {
                    Label("Draw new stratum", systemImage: "plus")
                }
                .accessibilityIdentifier("dashboard.drawStratumExtra")
            }
        } header: {
            Text("① Strata")
        } footer: {
            if !viewModel.strata.isEmpty {
                Text("Swipe left to delete. Area is computed automatically from lat/lon.")
            }
        }
    }

    // MARK: - Plan (step 2)

    private var planSection: some View {
        Section {
            NavigationLink {
                CruiseDesignScreen(project: viewModel.project)
            } label: {
                VStack(alignment: .leading, spacing: 2) {
                    Label("Cruise design", systemImage: "ruler")
                    Text("Choose plot size + sampling method → plots are generated automatically")
                        .font(.caption).foregroundStyle(.secondary)
                }
            }
            .accessibilityIdentifier("dashboard.designCruise")
            .disabled(viewModel.strata.isEmpty)

            NavigationLink {
                PlotMapScreen(project: viewModel.project)
            } label: {
                VStack(alignment: .leading, spacing: 2) {
                    Label("Plot map", systemImage: "map.fill")
                    Text("Review generated plot locations")
                        .font(.caption).foregroundStyle(.secondary)
                }
            }
            .accessibilityIdentifier("dashboard.plotMap")
        } header: {
            Text("② Cruise plan")
        } footer: {
            if viewModel.strata.isEmpty {
                Text("Register at least one stratum in ① first.")
            }
        }
    }

    // MARK: - Cruise (step 3)

    @ViewBuilder
    private var cruiseSection: some View {
        Section {
            if let design = viewModel.design {
                NavigationLink {
                    CruiseFlowScreen(project: viewModel.project,
                                     design: design)
                } label: {
                    VStack(alignment: .leading, spacing: 2) {
                        Label("Go Cruise",
                              systemImage: "figure.walk.circle.fill")
                            .font(.body.bold())
                        Text("Navigate to plot → record center → AR boundary → add trees")
                            .font(.caption).foregroundStyle(.secondary)
                    }
                }
                .accessibilityIdentifier("dashboard.goCruise")
            } else {
                HStack(alignment: .top, spacing: 8) {
                    Image(systemName: "lock.circle")
                        .foregroundStyle(.secondary)
                    VStack(alignment: .leading, spacing: 2) {
                        Text("Finish ② first")
                            .font(.subheadline.bold())
                        Text("You need a saved cruise design and generated plots before field measurement can begin.")
                            .font(.caption)
                            .foregroundStyle(.secondary)
                            .fixedSize(horizontal: false, vertical: true)
                    }
                }
            }
        } header: {
            Text("③ Field measurement")
        }
    }

    // MARK: - Tools (step 4 + misc)

    private var toolsSection: some View {
        Section {
            NavigationLink("Pre-field checklist") {
                PreFieldChecklistScreen(project: viewModel.project)
            }
            .accessibilityIdentifier("dashboard.preFieldCheck")
            NavigationLink("Calibrate this project") {
                CalibrationScreen(
                    viewModel: CalibrationViewModel(),
                    project: viewModel.project,
                    projectRepo: environment.projectRepository)
            }
            .accessibilityIdentifier("dashboard.calibrateProject")
            NavigationLink("Export results (PDF · CSV · GeoJSON)") {
                ExportScreen(project: viewModel.project)
            }
            .accessibilityIdentifier("dashboard.export")
            NavigationLink("Settings") {
                SettingsScreen()
            }
            .accessibilityIdentifier("dashboard.settings")
        } header: {
            Text("④ Tools")
        } footer: {
            Text("Calibration only needs to be done once — it noticeably improves accuracy.")
        }
    }

    // MARK: - Formatting

    private func formatAcres(_ value: Double) -> String {
        String(format: "%.2f acres", value)
    }

    #if canImport(UniformTypeIdentifiers)
    private var allowedTypes: [UTType] {
        switch importFormat {
        case .geoJSON:
            return [
                UTType(filenameExtension: "geojson") ?? .json,
                .json
            ]
        case .kml:
            return [UTType(filenameExtension: "kml") ?? .xml, .xml]
        }
    }
    #endif
}

File: android/app/src/main/java/com/hcjeong/forestix/ui/screens/project/ProjectDashboardScreen.kt
// Port of iOS Screens/ProjectDashboardScreen.swift.
// Spec §3.1 REQ-PRJ-002/003/004. Project dashboard — strata list,
// planned-plot summary, entry points into CruiseDesign / PlotMap /
// Export / Settings.
//
// Phase 7.4 redesign: whole screen reads top-to-bottom as a step-by-step
// guide (①→②→③→④). Each step shows its own friendly description and a
// primary action. The strata step supports both "draw on map" (no file
// needed) and legacy "import from file" paths.

package com.hcjeong.forestix.ui.screens.project

import android.widget.Toast
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.FileDownload
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Map
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.hcjeong.forestix.LocalAppEnvironment
import com.hcjeong.forestix.data.cruise.Project
import com.hcjeong.forestix.ui.Routes
import com.hcjeong.forestix.ui.clickableNoRipple
import com.hcjeong.forestix.ui.screens.ForestixScaffold
import com.hcjeong.forestix.ui.theme.Forestix
import com.hcjeong.forestix.ui.theme.ForestixSpace
import java.util.Locale
import java.util.UUID
import kotlinx.coroutines.launch

@Composable
fun ProjectDashboardScreen(nav: NavController, projectId: String) {
    val env = LocalAppEnvironment.current
    var project by remember { mutableStateOf<Project?>(null) }
    LaunchedEffect(projectId) {
        project = env.projectRepository.read(UUID.fromString(projectId))
    }
    val loaded = project
    if (loaded == null) {
        ForestixScaffold(nav, title = "Project") { padding ->
            Box(Modifier.padding(padding).fillMaxSize(), contentAlignment = Alignment.Center) {
                CircularProgressIndicator()
            }
        }
        return
    }
    ProjectDashboardContent(nav, loaded)
}

@Composable
private fun ProjectDashboardContent(nav: NavController, project: Project) {
    val env = LocalAppEnvironment.current
    val context = LocalContext.current
    val scope = rememberCoroutineScope()
    val colors = Forestix.colors
    val type = Forestix.type
    val viewModel = remember(project.id) { ProjectDashboardViewModel(project) }

    val strata by viewModel.strata.collectAsState()
    val plannedPlots by viewModel.plannedPlots.collectAsState()
    val design by viewModel.design.collectAsState()
    val closedPlotCount by viewModel.closedPlotCount.collectAsState()
    val errorMessage by viewModel.errorMessage.collectAsState()
    val toastMessage by viewModel.toastMessage.collectAsState()

    var importFormat by remember { mutableStateOf(ProjectDashboardViewModel.ImportFormat.GEO_JSON) }
    val importLauncher = rememberLauncherForActivityResult(
        ActivityResultContracts.OpenDocument(),
    ) { uri ->
        if (uri == null) return@rememberLauncherForActivityResult
        scope.launch {
            val bytes = try {
                context.contentResolver.openInputStream(uri)?.use { it.readBytes() }
            } catch (e: Exception) {
                viewModel.errorMessage.value = "Import failed: $e"
                null
            }
            if (bytes != null) viewModel.importStrata(bytes, importFormat)
        }
    }

    // iOS `.task` + `.onAppear` — runs on first appear and again each time
    // the dashboard re-enters composition (e.g. popping back from
    // CruiseDesign or PlotTally).
    LaunchedEffect(Unit) {
        viewModel.configure(env)
        viewModel.refresh()
    }

    LaunchedEffect(toastMessage) {
        val message = toastMessage ?: return@LaunchedEffect
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
        viewModel.toastMessage.value = null
    }

    ForestixScaffold(nav, title = project.name) { padding ->
        Column(
            Modifier
                .padding(padding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = ForestixSpace.md),
            verticalArrangement = Arrangement.spacedBy(ForestixSpace.md),
        ) {
            // MARK: - Getting started (progress guide)
            FormSection(header = "GUIDE") {
                Text("How this works", style = type.bodyBold, color = colors.textPrimary)
                StepRow(
                    n = 1, done = strata.isNotEmpty(),
                    title = "Define strata",
                    hint = "Draw boundaries on the map, or import GeoJSON / KML.")
                StepRow(
                    n = 2, done = design != null && plannedPlots.isNotEmpty(),
                    title = "Design cruise + generate plots",
                    hint = "Pick plot size and spacing — sample plots are generated automatically.")
                StepRow(
                    n = 3, done = closedPlotCount > 0,
                    title = "Measure in the field (Go Cruise)",
                    hint = "Walk to each plot and measure trees one by one. DBH via LiDAR, height via AR.")
                StepRow(
                    n = 4, done = plannedPlots.isNotEmpty() && closedPlotCount >= plannedPlots.size,
                    title = "Review + export",
                    hint = "Check stand statistics and export to PDF / CSV / GeoJSON.")
            }

            // MARK: - Summary
            FormSection(header = "Summary") {
                LabeledContentRow(
                    "Units",
                    project.units.raw.replaceFirstChar { it.uppercase(Locale.US) })
                LabeledContentRow("Total area", formatAcres(viewModel.totalAcres))
                LabeledContentRow("Strata", "${strata.size}")
                LabeledContentRow("Planned plots", "${plannedPlots.size}")
            }

            // MARK: - Strata (step 1)
            FormSection(
                header = "① Strata",
                footer = if (strata.isNotEmpty())
                    "Tap the trash icon to delete. Area is computed automatically from lat/lon."
                else null,
            ) {
                if (strata.isEmpty()) {
                    Row(verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(ForestixSpace.xs)) {
                        Icon(Icons.Filled.Map, contentDescription = null,
                            tint = colors.textPrimary, modifier = Modifier.size(16.dp))
                        Text("No strata yet", style = type.bodyBold, color = colors.textPrimary)
                    }
                    Text(
                        "A stratum is a cutting block you want to measure. Draw corners on the map, or import a prepared GeoJSON / KML file.",
                        style = type.caption, color = colors.textSecondary)
                    Button(
                        onClick = { nav.navigate(ProjectFlowRoutes.stratumDraw(project.id.toString())) },
                        modifier = Modifier.fillMaxWidth().height(44.dp),
                    ) {
                        Text("Draw stratum on map")
                    }
                    Box {
                        var importMenuOpen by remember { mutableStateOf(false) }
                        OutlinedButton(
                            onClick = { importMenuOpen = true },
                            modifier = Modifier.fillMaxWidth().height(40.dp),
                        ) {
                            Icon(Icons.Filled.FileDownload, contentDescription = null,
                                modifier = Modifier.size(16.dp))
                            Spacer(Modifier.size(ForestixSpace.xs))
                            Text("Import from file")
                        }
                        DropdownMenu(
                            expanded = importMenuOpen,
                            onDismissRequest = { importMenuOpen = false },
                        ) {
                            DropdownMenuItem(
                                text = { Text("Import GeoJSON") },
                                onClick = {
                                    importMenuOpen = false
                                    importFormat = ProjectDashboardViewModel.ImportFormat.GEO_JSON
                                    importLauncher.launch(arrayOf("application/json", "application/geo+json", "application/octet-stream", "*/*"))
                                },
                            )
                            DropdownMenuItem(
                                text = { Text("Import KML") },
                                onClick = {
                                    importMenuOpen = false
                                    importFormat = ProjectDashboardViewModel.ImportFormat.KML
                                    importLauncher.launch(arrayOf("application/vnd.google-earth.kml+xml", "application/xml", "text/xml", "*/*"))
                                },
                            )
                        }
                    }
                } else {
                    strata.forEach { stratum ->
                        Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
                            Column(Modifier.weight(1f)) {
                                Text(stratum.name, style = type.body, color = colors.textPrimary)
                                Text(formatAcres(stratum.areaAcres.toDouble()),
                                    style = type.caption, color = colors.textSecondary)
                            }
                            IconButton(onClick = { scope.launch { viewModel.delete(stratum.id) } }) {
                                Icon(Icons.Filled.Delete, contentDescription = "Delete",
                                    tint = colors.confidenceBad, modifier = Modifier.size(18.dp))
                            }
                        }
                    }
                    Row(
                        Modifier
                            .fillMaxWidth()
                            .clickableNoRipple { nav.navigate(ProjectFlowRoutes.stratumDraw(project.id.toString())) },
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(ForestixSpace.xs),
                    ) {
                        Icon(Icons.Filled.Add, contentDescription = null,
                            tint = colors.primary, modifier = Modifier.size(16.dp))
                        Text("Draw new stratum", style = type.body, color = colors.primary)
                    }
                }
            }

            // MARK: - Plan (step 2)
            FormSection(
                header = "② Cruise plan",
                footer = if (strata.isEmpty()) "Register at least one stratum in ① first." else null,
            ) {
                NavRow(
                    title = "Cruise design",
                    subtitle = "Choose plot size + sampling method → plots are generated automatically",
                    enabled = strata.isNotEmpty(),
                ) {
                    nav.navigate(ProjectFlowRoutes.cruiseDesign(project.id.toString()))
                }
                NavRow(
                    title = "Plot map",
                    subtitle = "Review generated plot locations",
                ) {
                    nav.navigate(ProjectFlowRoutes.plotMap(project.id.toString()))
                }
            }

            // MARK: - Cruise (step 3)
            FormSection(header = "③ Field measurement") {
                // Go Cruise unlocks only with BOTH a saved design AND
                // generated plots — otherwise CruiseFlowScreen hits the
                // "No planned plots yet" dead-end.
                if (design != null && plannedPlots.isNotEmpty()) {
                    NavRow(
                        title = "Go Cruise",
                        subtitle = cruiseSubtitle(plannedPlots.size, closedPlotCount),
                        bold = true,
                    ) {
                        nav.navigate(ProjectFlowRoutes.cruiseFlow(project.id.toString()))
                    }
                } else {
                    Row(horizontalArrangement = Arrangement.spacedBy(ForestixSpace.xs)) {
                        Icon(Icons.Filled.Lock, contentDescription = null,
                            tint = colors.textSecondary, modifier = Modifier.size(18.dp))
                        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                            Text("Finish ② first", style = type.bodyBold, color = colors.textPrimary)
                            Text(
                                "You need a saved cruise design and generated plots before field measurement can begin.",
                                style = type.caption, color = colors.textSecondary)
                        }
                    }
                }
            }

            // MARK: - Tools (step 4 + misc)
            FormSection(
                header = "④ Tools",
                footer = "Calibration only needs to be done once — it noticeably improves accuracy.",
            ) {
                NavRow(title = "Pre-field checklist") {
                    nav.navigate(ProjectFlowRoutes.preFieldChecklist(project.id.toString()))
                }
                NavRow(title = "Calibrate this project") {
                    nav.navigate(ProjectFlowRoutes.calibration(project.id.toString()))
                }
                NavRow(title = "Export results (PDF · CSV · GeoJSON)") {
                    nav.navigate(ProjectFlowRoutes.export(project.id.toString()))
                }
                NavRow(title = "Settings") {
                    nav.navigate(Routes.SETTINGS)
                }
            }

            Spacer(Modifier.height(ForestixSpace.xl))
        }
    }

    if (errorMessage != null) {
        AlertDialog(
            onDismissRequest = { viewModel.errorMessage.value = null },
            title = { Text("Something went wrong") },
            text = { Text(errorMessage ?: "") },
            confirmButton = {
                TextButton(onClick = { viewModel.errorMessage.value = null }) { Text("OK") }
            },
        )
    }
}

// MARK: - Helpers

/// Second line of the Go Cruise row — shows progress vs total so the
/// cruiser doesn't have to drill into the plot list to see how many
/// plots remain.
private fun cruiseSubtitle(total: Int, closed: Int): String {
    if (closed == 0) {
        return "Navigate to plot → record center → AR boundary → add trees"
    }
    return "$closed of $total plot${if (total == 1) "" else "s"} closed"
}

private fun formatAcres(value: Double): String =
    String.format(Locale.US, "%.2f acres", value)

// MARK: - Rows

@Composable
private fun StepRow(n: Int, done: Boolean, title: String, hint: String) {
    val colors = Forestix.colors
    val type = Forestix.type
    Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
        Box(
            Modifier
                .size(28.dp)
                .background(
                    if (done) colors.confidenceOk else colors.primaryMuted,
                    CircleShape),
            contentAlignment = Alignment.Center,
        ) {
            if (done) {
                Icon(Icons.Filled.Check, contentDescription = null,
                    tint = Color.White, modifier = Modifier.size(14.dp))
            } else {
                Text("$n", style = type.caption, color = colors.primary)
            }
        }
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(title, style = type.bodyBold, color = colors.textPrimary)
            Text(hint, style = type.caption, color = colors.textSecondary)
        }
    }
}

@Composable
private fun NavRow(
    title: String,
    subtitle: String? = null,
    enabled: Boolean = true,
    bold: Boolean = false,
    onClick: () -> Unit,
) {
    val colors = Forestix.colors
    val type = Forestix.type
    val alpha = if (enabled) 1f else 0.4f
    Row(
        Modifier
            .fillMaxWidth()
            .clickableNoRipple { if (enabled) onClick() },
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                title,
                style = if (bold) type.bodyBold else type.body,
                color = colors.textPrimary.copy(alpha = alpha))
            if (subtitle != null) {
                Text(subtitle, style = type.caption,
                    color = colors.textSecondary.copy(alpha = alpha))
            }
        }
        Icon(Icons.AutoMirrored.Filled.KeyboardArrowRight, contentDescription = null,
            tint = colors.textTertiary, modifier = Modifier.size(14.dp))
    }
}

/// Suppress unused-import style warnings for FontStyle if the italic
/// species row moves; kept because CruiseDesignScreen shares this package.
private val unusedFontStyle = FontStyle.Italic
